src/components/cards/TradeCard/TradePair.vue

```diff
@@ -1,373 +1,161 @@
 import useTokens from '@/composables/useTokens';
-import { NATIVE_ASSET_ADDRESS } from '@/constants/tokens';
-
-const ETH_BUFFER = 0.1;
-
-export default defineComponent({
-  components: {
-    TradePairToggle,
-    SelectTokenModal
-  },
-  props: {
-    tokenInAmountInput: {
-      type: String,
-      required: true
-    },
-    tokenInAddressInput: {
-      type: String,
-      required: true
-    },
-    tokenOutAmountInput: {
-      type: String,
-      required: true
-    },
-    tokenOutAddressInput: {
-      type: String,
-      required: true
-    },
-    exactIn: {
-      type: Boolean,
-      required: true
-    },
-    priceImpact: {
-      type: Number,
-      required: true
-    }
-  },
-  emits: [
-    'tokenInAddressChange',
-    'tokenInAmountChange',
-    'tokenOutAddressChange',
-    'tokenOutAmountChange',
-    'exactInChange',
-    'change'
-  ],
-  setup(props, { emit }) {
-    const { tokens, balances, injectTokens } = useTokens();
-    const { fNum, toFiat } = useNumbers();
-
-    const {
-      tokenInAmountInput,
-      tokenInAddressInput,
-      tokenOutAmountInput,
-      tokenOutAddressInput,
-      exactIn
-    } = toRefs(props);
-
-    const tokenInValue = computed(() =>
-      toFiat(tokenInAmountInput.value, tokenInAddressInput.value)
-    );
-
-    const tokenInSymbol = computed(() => {
-      const tokenIn = tokens.value[tokenInAddressInput.value];
-      const symbol = tokenIn ? tokenIn.symbol : '';
-      return symbol;
-    });
-
-    const tokenOutValue = computed(() =>
-      toFiat(tokenOutAmountInput.value, tokenOutAddressInput.value)
-    );
-
-    const tokenOutSymbol = computed(() => {
-      const tokenOut = tokens.value[tokenOutAddressInput.value];
-      const symbol = tokenOut ? tokenOut.symbol : '';
-      return symbol;
-    });
-
-    const tokenInDecimals = computed(() => {
-      const decimals = tokens.value[tokenInAddressInput.value]
-        ? tokens.value[tokenInAddressInput.value].decimals
-        : 18;
-      return decimals;
-    });
-
-    const tokenOutDecimals = computed(() => {
-      const decimals = tokens.value[tokenOutAddressInput.value]
-        ? tokens.value[tokenOutAddressInput.value].decimals
-        : 18;
-      return decimals;
-    });
-
-    const isInRate = ref(true);
-    const modalSelectTokenType = ref('input');
-    const modalSelectTokenIsOpen = ref(false);
-
-    function handleMax(): void {
-      const balance = balances.value[tokenInAddressInput.value] || '0';
-      const balanceNumber = parseFloat(balance);
-      const maxAmount =
-        tokenInAddressInput.value !== NATIVE_ASSET_ADDRESS
-          ? balance
-          : balanceNumber > ETH_BUFFER
-          ? (balanceNumber - ETH_BUFFER).toString()
-          : '0';
-      handleInAmountChange(maxAmount);
-    }
-
-    function handleInAmountChange(value: string): void {
-      emit('exactInChange', true);
-      emit('tokenInAmountChange', value);
-      emit('change', value);
-    }
```

Comment:
A lot of this logic is now handled by the TokenInput component. One thing I might not have handled correctly is setting the max in the Gnosis case.